### PR TITLE
Feature/testing release 0.1.0

### PR DESCRIPTION
**Pull request by tdesfont**
Add unit tests runnable through make unit-tests in the dss-plugin... root repository

### CODE REVIEW

File: tests/python/unit/test_base.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    # Load data into vector format for indexing
+    columns = [params["unique_id_column"]] + params["feature_columns"]
+    input_df = pd.read_csv('./tests/resources/caltech_embeddings.csv')
+    # Restrict to selected columns
+    input_df = input_df[columns]
+    data_loader = DataLoader(params["unique_id_column"], params["feature_columns"])
+    (vector_ids, vectors) = data_loader.convert_df_to_vectors(input_df)
+    nearest_neighbor = NearestNeighborSearch(num_dimensions=vectors.shape[1], **params)
+    tmp = NamedTemporaryFile()
```
Comment (alexcombessie) [inline tests/python/unit/test_base.py:26]: Usually, it's better practice to use a `with` statement

File: tests/python/unit/test_base.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    'expert': True}
+
+    # Load data into vector format for indexing
+    columns = [params["unique_id_column"]] + params["feature_columns"]
+    input_df = pd.read_csv('./tests/resources/caltech_embeddings.csv')
+    input_df = input_df[columns]
+    data_loader = DataLoader(params["unique_id_column"], params["feature_columns"])
+    (vector_ids, vectors) = data_loader.convert_df_to_vectors(input_df)
+    nearest_neighbor = NearestNeighborSearch(num_dimensions=vectors.shape[1], **params)
+    tmp = NamedTemporaryFile()
```
Comment (alexcombessie) [inline tests/python/unit/test_base.py:57]: Same comment, maybe better to do a `with` statement

File: tests/python/unit/test_base.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    nearest_neighbor.build_save_index(vectors=vectors, index_path=tmp.name)
+    file_exist = os.path.isfile(tmp.name)
+    tmp.close()
+    # Test if file is created as a result
+    assert file_exist
```
Comment (alexcombessie) [inline tests/python/unit/test_base.py:31]: Maybe better to `assert` right after `file_exist`? Or just `assert os.path.isfile(tmp.name)` directly?

**Review by alexcombessie — CHANGES_REQUESTED** [review]
The tests all pass, which is great! Good job.

 Apart from minor suggestions I left in the code, my only additional request would be to remove all the `__init__.py`. If I understand correctly [this doc](https://docs.pytest.org/en/stable/goodpractices.html) this is not needed in our case.

**Comment by alexcombessie** [comment]
Related issue: there tests are failing on Jenkins: http://dku34.dataiku.com:13456/jenkins/job/Dataiku/job/dss-plugin-similarity-search/view/change-requests/job/PR-3/lastBuild/testReport/junit/tests.python.unit/test_base/test_build_save_index/

Can you investigate?

File: tests/python/unit/test_base.py (hunk 3 above)
Comment (tdesfont) [reply tests/python/unit/test_base.py]: Done

File: tests/python/unit/test_base.py (hunk 2 above)
Comment (tdesfont) [reply tests/python/unit/test_base.py]: Done

File: tests/python/unit/test_base.py (hunk 1 above)
Comment (tdesfont) [reply tests/python/unit/test_base.py]: Done

**Review by alexcombessie — CHANGES_REQUESTED** [review]
All good. The unit tests pass successfully on my mac. For some strange reason, they still fail on the dku34 Jenkins
http://dku34.dataiku.com:13456/jenkins/job/Dataiku/job/dss-plugin-similarity-search/view/change-requests/job/PR-3/3/testReport/junit/tests.python.unit/test_base/test_build_save_index/

Once that last thing is solved, we can merge.

**Review by alexcombessie — APPROVED** [review]
Super! Let's merge

**Pull request merged**